test-kit/src/test/scala/format/BinaryFenTest.scala

```diff
@@ -7,31 +7,31 @@ import munit.ScalaCheckSuite
 import org.scalacheck.Gen
 import org.scalacheck.Prop.forAll
 
-import scala.collection.mutable.ListBuffer
+import scala.collection.mutable.ArrayBuilder
 
 class BinaryFenTest extends ScalaCheckSuite:
   test("long roundtrip"):
     forAll: (v: Long) =>
-      val builder = new ListBuffer[Byte]()
+      val builder = ArrayBuilder.ofByte()
       BinaryFen.implementation.writeLong(builder, v)
       assertEquals(BinaryFen.implementation.readLong(builder.result.iterator), v)
 
   test("leb128 roundtrip"):
     forAll(Gen.posNum[Int]): (v: Int) =>
-      val builder = new ListBuffer[Byte]()
+      val builder = ArrayBuilder.ofByte()
       BinaryFen.implementation.writeLeb128(builder, v)
       assertEquals(BinaryFen.implementation.readLeb128(builder.result.iterator), v)
 
   private val genNibble = Gen.chooseNum[Int](0, 15)
 
   test("nibbles roundtrip"):
     forAll(genNibble, genNibble): (lo: Int, hi: Int) =>
-      val builder = new ListBuffer[Byte]()
+      val builder = ArrayBuilder.ofByte()
       BinaryFen.implementation.writeNibbles(builder, lo, hi)
       assertEquals(BinaryFen.implementation.readNibbles(builder.result.iterator), (lo, hi))
 
   test("rewrite fixpoint"):
-    forAll: (bytes: List[Byte]) =>
+    forAll: (bytes: Array[Byte]) =>
       val rewritten = BinaryFen.write(BinaryFen(bytes).read)
       assertEquals(BinaryFen.write(rewritten.read), rewritten)
 
```

Comment:
```suggestion

  test("equals are sensiable"):
    forAll: (bytes: Array[Byte]) =>
      val anotherByes = bytes.clone()
      assertEquals(BinaryFen(bytes), BinaryFen(anotherByes))
```

Add a test to make sure `equals` are sensible